### PR TITLE
ci: Add FOSSA license checking

### PR DESCRIPTION
**Pull request by russellb**
During CNCF onboarding, we were advised to adopt an automatic license
checking tool. This PR enables running FOSSA against MetalLB PRs.

issue #1060

### CODE REVIEW

File: .github/workflows/fossa.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  fossa:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: fossa-contrib/fossa-action@v1
+        with:
+          # https://docs.fossa.com/docs/api-reference#push-only-api-token
+          fossa-api-key: 831247a6ab6f0d4f99879e0353b512a5
```
Comment (msherif1234) [inline .github/workflows/fossa.yaml:16]: is this key statically assigned to us some how ?

Comment (russellb) [reply .github/workflows/fossa.yaml:16]: It's a push-only API key I created.  That's what the docs said to do for the github action

https://github.com/marketplace/actions/fossa-action

Comment (msherif1234) [reply .github/workflows/fossa.yaml:16]: do u need to plugin somethig to config.yaml for this to be part of the CI ?

Comment (msherif1234) [reply .github/workflows/fossa.yaml:16]: there is also fossa analyze in this example https://github.com/k8ssandra/cass-operator/blob/master/.github/workflows/license-check.yml

Comment (fedepaol) [reply .github/workflows/fossa.yaml:16]: This is for github actions.
Here I guess it's using a circleCI orb.

**Comment by fedepaol** [comment]
I can see the new lane working.
LGTM

**Pull request merged**